src/installer/corehost/cli/hostpolicy/coreclr.cpp

```diff
@@ -5,72 +5,28 @@
 #include <cassert>
 
 #include "coreclr.h"
+#include "coreclr_resolver_t.h"
 #include <utils.h>
 #include <error_codes.h>
 
-// Prototype of the coreclr_initialize function from coreclr.dll
-using coreclr_initialize_fn = pal::hresult_t(STDMETHODCALLTYPE *)(
-    const char* exePath,
-    const char* appDomainFriendlyName,
-    int propertyCount,
-    const char** propertyKeys,
-    const char** propertyValues,
-    coreclr_t::host_handle_t* hostHandle,
-    unsigned int* domainId);
-
-// Prototype of the coreclr_shutdown function from coreclr.dll
-using coreclr_shutdown_fn = pal::hresult_t(STDMETHODCALLTYPE *)(
-    coreclr_t::host_handle_t hostHandle,
-    unsigned int domainId,
-    int* latchedExitCode);
-
-// Prototype of the coreclr_execute_assembly function from coreclr.dll
-using coreclr_execute_assembly_fn = pal::hresult_t(STDMETHODCALLTYPE *)(
-    coreclr_t::host_handle_t hostHandle,
-    unsigned int domainId,
-    int argc,
-    const char** argv,
-    const char* managedAssemblyPath,
-    unsigned int* exitCode);
-
-// Prototype of the coreclr_create_delegate function from coreclr.dll
-using coreclr_create_delegate_fn = pal::hresult_t(STDMETHODCALLTYPE *)(
-    coreclr_t::host_handle_t hostHandle,
-    unsigned int domainId,
-    const char* entryPointAssemblyName,
-    const char* entryPointTypeName,
-    const char* entryPointMethodName,
-    void** delegate);
-
 namespace
 {
-    pal::dll_t g_coreclr = nullptr;
     coreclr_shutdown_fn coreclr_shutdown = nullptr;
     coreclr_initialize_fn coreclr_initialize = nullptr;
     coreclr_execute_assembly_fn coreclr_execute_assembly = nullptr;
     coreclr_create_delegate_fn coreclr_create_delegate = nullptr;
 
     bool coreclr_bind(const pal::string_t& libcoreclr_path)
     {
-        assert(g_coreclr == nullptr);
-
-        pal::string_t coreclr_dll_path(libcoreclr_path);
-        append_path(&coreclr_dll_path, LIBCORECLR_NAME);
-
-        if (!pal::load_library(&coreclr_dll_path, &g_coreclr))
-        {
-            return false;
-        }
+        assert(coreclr_initialize == nullptr);
 
-        coreclr_initialize = reinterpret_cast<coreclr_initialize_fn>(pal::get_symbol(g_coreclr, "coreclr_initialize"));
-        coreclr_shutdown = reinterpret_cast<coreclr_shutdown_fn>(pal::get_symbol(g_coreclr, "coreclr_shutdown_2"));
-        coreclr_execute_assembly = reinterpret_cast<coreclr_execute_assembly_fn>(pal::get_symbol(g_coreclr, "coreclr_execute_assembly"));
-        coreclr_create_delegate = reinterpret_cast<coreclr_create_delegate_fn>(pal::get_symbol(g_coreclr, "coreclr_create_delegate"));
+        coreclr_resolver_contract_t contract;
```

Comment:
I would store the contract itself instead of extracting the function pointers from it - it's shorter and for some reason I just like it better.

Reply:
I felt the same way, but want to keep code changes to minimum. I will try to store the contract.